### PR TITLE
DINEOFs/compilation bug fixes

### PR DESCRIPTION
**Pull request by aburrell**
# Description

This pull request contains bug fixes needed to:
- compile/install the module
- use the pydineofs_dineofs sub-module

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Installling the module locally and loading a pydineofs test instrument

```
import datetime as dt
import pysat
import pysatModels

dineofs = pysat.Instrument(inst_module=pysatModels.models.pydineof_dineof, tag='test')
stime = dt.datetime(2009, 1, 1)
dineofs.download(start=stime)
dineofs.load(date=stime)
print(dineofs)
```

Yeilds:
```
pysat Instrument object
-----------------------
Platform: 'pydineof'
Name: 'dineof'
Tag: 'test'
Instrument id: ''

Data Processing
---------------
Cleaning Level: 'clean'
Data Padding: None
Keyword Arguments Passed to list_files: {}
Keyword Arguments Passed to load: {}
Keyword Arguments Passed to preprocess: {}
Keyword Arguments Passed to download: {}
Keyword Arguments Passed to list_remote_files: {}
Keyword Arguments Passed to clean: {}
Keyword Arguments Passed to init: {}
Custom Functions: 0 applied

Local File Statistics
---------------------
Number of files: 1
Date Range: 01 January 2009 --- 01 January 2009

Loaded Data Statistics
----------------------
Date: 01 January 2009
DOY: 001
Time range: 01 January 2009 00:00:00 --- 01 January 2009 00:00:00
Number of Times: 1
Number of variables: 21

Variable Names:
testing_1_coord_0    testing_1_coord_1    time                 
                              ...                              
testing_2_basis_mean testing_2_basis_norm basis_amplitude      

pysat Meta object
-----------------
Tracking 11 metadata values
Metadata for 21 standard variables
Metadata for 0 ND variables
```

**Test Configuration**:
* Operating system: OS X Mojave
* Version number: 3.7
* Any details about your local setup that are relevant: using ipython, main pysat branch

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Review by aburrell — COMMENTED** [review]
Remember, DO NOT delete `match_doc` branch after merging.  The main purpose of this branch has not been met.

File: docs/examples/ex_match.rst
Comment (rstoneback) [inline docs/examples/ex_match.rst:20]:
```suggestion
to the model time rather than all observations within the model time.
```

**Review by rstoneback — COMMENTED** [review]
Changes look good to me. Awaiting feedback on uploaded test file to ensure it is in good shape. Waiting since example includes test file and if the file changes the example could change.

File: docs/examples/ex_match.rst
Comment (rstoneback) [inline docs/examples/ex_match.rst:35]: Missing pysatMadrigal import

File: docs/examples/ex_match.rst
Comment (rstoneback) [inline docs/examples/ex_match.rst:54]:
```suggestion
   dmsp_f18 = pysat.Instrument(inst_module=pysatMadrigal.instruments.dmsp_ivm, inst_id='f18', clean_level='none')
```

Comment (rstoneback) [reply docs/examples/ex_match.rst:54]: Also missing `user` and `password` in dmsp download line. Won't let me comment directly on that one. The error feedback without `user` etc. is good though.

Comment (rstoneback) [reply docs/examples/ex_match.rst:54]: dmsp_f18.custom_attach

Comment (rstoneback) [reply docs/examples/ex_match.rst:54]: sami2_meridian no longer defined.

Comment (rstoneback) [reply docs/examples/ex_match.rst:54]: There is no latitude in the DINEOF file and 'inst_lon_name' is 'glon'.

**Comment by aburrell** [comment]
Ok, this looks ready to merge if you approve @rstoneback.  Then I can get to writing the example on this branch, while develop has a functioning dineofs instrument.

File: docs/examples/ex_match.rst
Comment (rstoneback) [reply docs/examples/ex_match.rst:54]: I think I'm starting to get the # Ended here comment :)

**Review by rstoneback — COMMENTED** [review]
I tried going through the example, listed the things I encountered along the way, and then I think I sorted out why you had the # ended here comment.

**Review by rstoneback — APPROVED** [review]

**Pull request merged**
